[MailmapInfo] Ignore case for email address

slietzau opened the pull request:
`git shortlog` ignores the case when matching the email address. As such, `git filter-repo` should do the same.

newren commented:
Thanks for the contribution, and the good explanation in the commit message.  Sorry for taking so long to get back to you.  A couple things:
  * Could you add a signed-off-by line to your commit message (`git commit --amend -s` should do it) as per contributing guidelines?
  * Could you add a testcase which would have failed under the old code but passes with the new rule?  Or, since I took so long to respond, explain a testcase in words in this pull request and I'll go and add one based on that.

newren reviewed (changes requested):
(Just adding an extra comment so that this time I can mark it as request changes.)

newren commented:
Can't merge without the Signed-off-by, so I'll go ahead and close.  Feel free to reopen if you're willing to change the commit message to include your signed-off-by.

slietzau commented:
@newren Sorry this took so long. I updated the commit message with the "Signed-off-by" and rebased to the latest main branch. If you are willing to merge, you can reopen the PR. I can also make a new one if desired.

newren commented:
There was a logic error causing the tests to fail.  I fixed it up, tweaked the commit message and merged as commit c9a9dcc886e7 (filter-repo: ignore case for email address with mailmap, 2020-09-11).

Thanks for the contribution!
